File: 20-valid-parentheses/20-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        List<Character> ch = Arrays.asList('(', '{', '[');
        Stack<Character> st = new Stack<>();
        for(int i=0;i<s.length();i++){
            if(ch.contains(s.charAt(i)))
                st.push(s.charAt(i));
            else if(s.charAt(i) == '}' && !st.isEmpty() && st.peek() != '{')
                return false;
            else if(s.charAt(i) == ')' && !st.isEmpty() && st.peek() != '(')
                return false;
            else if(s.charAt(i) == ']' && !st.isEmpty() && st.peek() != '[')
                return false;
            else if(st.isEmpty())
                return false;
            else
                st.pop();  
        }
        return st.isEmpty();
    }
}

File: 20-valid-parentheses/20-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> st;
        int n = s.length();
        char arr[] = {'{', '[', '('};
        for(int i=0;i<n;i++){
            if(!st.empty() && s[i] == ')' && st.top() != '(')
                return false;
            else if(!st.empty() && s[i] == ']' && st.top() != '[')
                return false;
            else if(!st.empty() && s[i] == '}' && st.top() != '{')
                return false;
            else if(find(begin(arr), end(arr), s[i]) != end(arr))
                st.push(s[i]);
            else if (st.empty())
                return false;
            else st.pop();
            
        }
        return st.empty();
    }
};